suite/src/main.rs

```diff
@@ -0,0 +1,30 @@
+#![no_std]
+#![no_main]
+#![feature(custom_test_frameworks)]
+#![feature(default_alloc_error_handler)]
+#![test_runner(crate::runtime::test_runner)]
+#![reexport_test_harness_main = "test_main"]
+
+mod modules;
+mod platform;
+#[macro_use]
+mod runtime;
+
+use platform::Platform;
+use riscv_rt::entry;
+
+extern crate alloc;
+
+fn main() {}
```

Comment:
Would it better to loop infinitely?

Reply:
That is one possibility, and the final suite will definitely have a big loop where it waits for commands and then executes them. 

The suite in the current state was written in a way to let the platform decide what to do once the code finishes, that is the purpose of the `platform::current().suspend(exit_code)` function.
For example on the qemu platform the current implementation uses a debug device provided by the emulator to stop the simulation process.
This is especially useful when running tests.

Reply:
Ok, makes sense.